hw/top_earlgrey/ip/sensor_ctrl/data/sensor_ctrl.hjson

```diff
@@ -0,0 +1,166 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+// TODO: This module is only a draft implementation that covers most of the ast_wrapper
+// functionality but is incomplete
+
+
+# SENSOR_CTRL register template
+#
+{
+  name: "SENSOR_CTRL",
+  clock_primary: "clk_i",
+  bus_device: "tlul",
+  regwidth: "32",
+  param_list: [
+    { name:    "NumAlerts",
+      type:    "int",
+      default: "7",
+      desc:    "Number of alerts",
+      local:   "true"
+    },
+    { name:    "NumIoRails",
+      type:    "int",
+      default: "2",
+      desc:    "Number of IO rails",
+      local:   "true"
+    }
+  ],
+
+  alert_list: [
+    { name: "ast_alerts",
+      width: "7", // need to update tool to support param
```

Comment:
initially I imagined this to be used rather like the `interrupt_list`, i.e., each alert is specified explicitly with its own name and description.

this multi-bit alert can work as well, but I think this needs a change to `topgen` to correctly recognize multi-bit alerts, here:
https://github.com/lowRISC/opentitan/blob/eb333ec2362ade4f920e9b3f9ebf8ec28aa191a2/util/topgen.py#L141-L144
 
e.g., something like
```python
        async_on = ""
        for alert in top['alert']:
            for k in alert['width']:
                async_on = str(alert['async']) + async_on
        async_on = ("%d'b" % n_alerts) + async_on
```

Reply:
never mind this comment, I saw that you already fixed this in `topgen` below